bundle/Service/ScheduledVisibilityService.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Netgen\Bundle\IbexaScheduledVisibilityBundle\Service;
+
+use DateTime;
+use Ibexa\Contracts\Core\Repository\Values\Content\Content;
+use Ibexa\Contracts\Core\Repository\Values\Content\Field;
+use Ibexa\Core\FieldType\Date\Value as DateValue;
+use Ibexa\Core\FieldType\DateAndTime\Value as DateAndTimeValue;
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\Enums\HandlerType;
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\Enums\VisibilityUpdateResult;
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\ScheduledVisibility\ScheduledVisibilityInterface;
+use OutOfBoundsException;
+
+use function in_array;
+use function sprintf;
+
+final class ScheduledVisibilityService
+{
+    public function __construct(
+        private readonly iterable $handlers,
+        private readonly string $type,
+        private readonly bool $enabled,
```

Comment:
Let's extract configuration handling into a separate service. That will enable us easier handling of the configuration where it's needed (command, event, and here) and will also simplify the implementation here and enable usage of the `ScheduledVisibilityService` regardless of the configuration, if needed.

Reply:
fixed in 45c89574595455ebecfd170915926b5d05df1a60

Reply:
Great 👍